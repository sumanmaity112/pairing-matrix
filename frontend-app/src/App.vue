<template>
  <div id="container">
    <PairingMatrixPage />
  </div>
</template>

<script>
import PairingMatrixPage from "@/components/PairingMatrixPage";

export default {
  name: "App",
  components: {
    PairingMatrixPage,
  },
};
</script>
