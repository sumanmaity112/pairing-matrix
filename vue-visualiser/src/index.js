import ChordPairingChart from "./components/ChordPairingChart";

export { ChordPairingChart };
