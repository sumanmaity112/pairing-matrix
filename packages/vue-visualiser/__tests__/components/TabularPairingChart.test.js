import { mount } from "@vue/test-utils";
import { pairingMatrix } from "../data/testData";
import { TabularPairingChart } from "../../src/index";

describe("TabularPairingChart", () => {
  afterEach(() => {
    document.body.innerHTML = "";
  });

  it("should render tabular pairing chart", async () => {
    const wrapper = mount(TabularPairingChart, {
      props: {
        height: 700,
        width: 900,
        authors: pairingMatrix.authors,
        data: pairingMatrix.matrix,
      },
      attachTo: document.body,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should refresh tabular pairing chart on data update", async () => {
    const initialData = pairingMatrix.matrix;
    const newData = {
      author: "joe",
      coAuthor: "tonystark",
      times: 1,
    };
    const updatedData = [...initialData, newData];
    const authors = pairingMatrix.authors;

    const wrapper = mount(TabularPairingChart, {
      props: { height: 700, width: 900, authors: authors, data: initialData },
      attachTo: document.body,
    });

    expect(wrapper.html()).toMatchSnapshot();

    await wrapper.setProps({
      data: updatedData,
      authors: [...authors, newData.author],
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
