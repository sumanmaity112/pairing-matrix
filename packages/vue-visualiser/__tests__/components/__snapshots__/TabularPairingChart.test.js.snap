// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabularPairingChart should refresh tabular pairing chart on data update 1`] = `
<div id="tabular-pairing-chart"><svg viewBox="-900,-700,900,700" svg-for="pairing-matrix-tabular-chart">
    <g location="top"><text co-author="zoe.morgan" transform="translate( -609.1666666666666 , -550) rotate(270)" font-size="18" font-family="sans-serif">zoe.morgan</text><text co-author="tonystark" transform="translate( -367.50000000000006 , -550) rotate(270)" font-size="18" font-family="sans-serif">tonystark</text><text co-author="john.reese" transform="translate( -125.83333333333336 , -550) rotate(270)" font-size="18" font-family="sans-serif">john.reese</text></g>
    <g location="side">
      <g transform="translate(-725, -525)"><text author="john.reese" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">john.reese</text>
        <g author="john.reese">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="tonystark" x="241.66666666666666" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="john.reese" x="483.3333333333333" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="john.reese"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">2</text><text co-author="tonystark" x="352.5" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">1</text><text co-author="john.reese" x="594.1666666666666" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -350)"><text author="tonystark" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">tonystark</text>
        <g author="tonystark">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="tonystark" x="241.66666666666666" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="tonystark"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">-</text><text co-author="tonystark" x="352.5" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -175)"><text author="zoe.morgan" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">zoe.morgan</text>
        <g author="zoe.morgan">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="zoe.morgan"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
    </g>
  </svg></div>
`;

exports[`TabularPairingChart should refresh tabular pairing chart on data update 2`] = `
<div id="tabular-pairing-chart"><svg viewBox="-900,-700,900,700" svg-for="pairing-matrix-tabular-chart">
    <g location="top"><text co-author="zoe.morgan" transform="translate( -639.375 , -550) rotate(270)" font-size="18" font-family="sans-serif">zoe.morgan</text><text co-author="tonystark" transform="translate( -458.125 , -550) rotate(270)" font-size="18" font-family="sans-serif">tonystark</text><text co-author="john.reese" transform="translate( -276.875 , -550) rotate(270)" font-size="18" font-family="sans-serif">john.reese</text><text co-author="joe" transform="translate( -95.625 , -550) rotate(270)" font-size="18" font-family="sans-serif">joe</text></g>
    <g location="side">
      <g transform="translate(-725, -525)"><text author="joe" x="-25" y="60.625" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">joe</text>
        <g author="joe">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="tonystark" x="181.25" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="john.reese" x="362.5" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="joe" x="543.75" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
        </g>
        <g author="joe"><text co-author="zoe.morgan" x="80.625" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">-</text><text co-author="tonystark" x="261.875" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">1</text><text co-author="john.reese" x="443.125" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">-</text><text co-author="joe" x="624.375" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -393.75)"><text author="john.reese" x="-25" y="60.625" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">john.reese</text>
        <g author="john.reese">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="tonystark" x="181.25" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="john.reese" x="362.5" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
        </g>
        <g author="john.reese"><text co-author="zoe.morgan" x="80.625" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">2</text><text co-author="tonystark" x="261.875" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">1</text><text co-author="john.reese" x="443.125" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -262.5)"><text author="tonystark" x="-25" y="60.625" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">tonystark</text>
        <g author="tonystark">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
          <rect co-author="tonystark" x="181.25" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
        </g>
        <g author="tonystark"><text co-author="zoe.morgan" x="80.625" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">-</text><text co-author="tonystark" x="261.875" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -131.25)"><text author="zoe.morgan" x="-25" y="60.625" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">zoe.morgan</text>
        <g author="zoe.morgan">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="171.25" height="121.25"></rect>
        </g>
        <g author="zoe.morgan"><text co-author="zoe.morgan" x="80.625" y="60.625" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
    </g>
  </svg></div>
`;

exports[`TabularPairingChart should render tabular pairing chart 1`] = `
<div id="tabular-pairing-chart"><svg viewBox="-900,-700,900,700" svg-for="pairing-matrix-tabular-chart">
    <g location="top"><text co-author="zoe.morgan" transform="translate( -609.1666666666666 , -550) rotate(270)" font-size="18" font-family="sans-serif">zoe.morgan</text><text co-author="tonystark" transform="translate( -367.50000000000006 , -550) rotate(270)" font-size="18" font-family="sans-serif">tonystark</text><text co-author="john.reese" transform="translate( -125.83333333333336 , -550) rotate(270)" font-size="18" font-family="sans-serif">john.reese</text></g>
    <g location="side">
      <g transform="translate(-725, -525)"><text author="john.reese" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">john.reese</text>
        <g author="john.reese">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="tonystark" x="241.66666666666666" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="john.reese" x="483.3333333333333" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="john.reese"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">2</text><text co-author="tonystark" x="352.5" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">1</text><text co-author="john.reese" x="594.1666666666666" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -350)"><text author="tonystark" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">tonystark</text>
        <g author="tonystark">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
          <rect co-author="tonystark" x="241.66666666666666" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="tonystark"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">-</text><text co-author="tonystark" x="352.5" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
      <g transform="translate(-725, -175)"><text author="zoe.morgan" x="-25" y="82.5" text-anchor="end" alignment-baseline="middle" font-size="18" font-family="sans-serif">zoe.morgan</text>
        <g author="zoe.morgan">
          <rect co-author="zoe.morgan" x="0" style="outline: 2px solid chocolate; border-radius: 6px;" fill="white" width="231.66666666666666" height="165"></rect>
        </g>
        <g author="zoe.morgan"><text co-author="zoe.morgan" x="110.83333333333333" y="82.5" alignment-baseline="middle" font-size="18" font-family="sans-serif">:)</text></g>
      </g>
    </g>
  </svg></div>
`;
